sunkit_image/tests/test_radial.py

```diff
@@ -193,6 +196,84 @@ def test_fig_fnrgf(smap):
 
     out.plot()
 
+@figure_test
+@pytest.mark.remote_data()
+def test_fig_rhef(smap):
+    radial_bin_edges = utils.equally_spaced_bins(0, 2, smap.data.shape[1])
+    radial_bin_edges *= u.R_sun
+
+    # Get the SDO AIA 171 colormap
+    aia_171_colormap=matplotlib.colormaps['sdoaia171']
+
+    # Assuming rad and smap are already defined
+    out_map1 = rad.rhef(smap, radial_bin_edges=radial_bin_edges, upsilon=None, method="numpy")
+    out_map2 = rad.rhef(smap, radial_bin_edges=radial_bin_edges, upsilon=(0.5, 0.5), method="scipy")
+    out_map3 = rad.rhef(smap, radial_bin_edges=radial_bin_edges, upsilon=(0.5, 0.25), method="numpy")
+    out_map4 = rad.rhef(smap, radial_bin_edges=radial_bin_edges, upsilon=(0.25, 0.5), method="numpy")
+    out_map5 = rad.rhef(smap, radial_bin_edges=radial_bin_edges, upsilon=(0.4, 0.4), method="scipy")
+
+    # Extract the data from the maps
+    data0 = np.log10(smap.data-np.nanmin(smap.data))**2
+    data1 = out_map1.data
+    data2 = out_map2.data
+    data3 = out_map3.data
+    data4 = out_map4.data
+    data5 = out_map5.data
+
+    # Get the coordinate ranges from the meta information
+    x_coords = out_map1.meta['cdelt1'] * (out_map1.data.shape[1] // 2)
+    y_coords = out_map1.meta['cdelt2'] * (out_map1.data.shape[0] // 2)
+    extent = [-x_coords, x_coords, -y_coords, y_coords]
+
+    # Create a figure with six subplots
+    fig, axs = plt.subplots(2, 3, figsize=(15, 10), sharey='all', sharex='all')
+    axs = axs.flatten()
+
+    # Plot the original map
+    im0 = axs[0].imshow(data0, origin='lower', extent=extent, cmap=aia_171_colormap)
+    axs[0].set_title("Log10(data)^2")
+
+    # Plot the first map
+    im1 = axs[1].imshow(data1, origin='lower', extent=extent, cmap=aia_171_colormap)
+    axs[1].set_title("Upsilon = None")
+
+    # Plot the second map
+    im2 = axs[2].imshow(data2, origin='lower', extent=extent, cmap=aia_171_colormap)
+    axs[2].set_title("Upsilon = (0.5, 0.5)")
+
+    # Plot the third map
+    im3 = axs[3].imshow(data3, origin='lower', extent=extent, cmap=aia_171_colormap)
+    axs[3].set_title("Upsilon = (0.5, 0.25)")
+
+    # Plot the fourth map
+    im2 = axs[4].imshow(data4, origin='lower', extent=extent, cmap=aia_171_colormap)
+    axs[4].set_title("Upsilon = (0.25, 0.5)")
+
+    # Plot the fifth map
+    im3 = axs[5].imshow(data5, origin='lower', extent=extent, cmap=aia_171_colormap)
+    axs[5].set_title("Upsilon = (0.4, 0.4)")
+
+
+    # Adjust layout
+    plt.tight_layout()
+
+    # Show the plot
+    plt.show()
+
+    # # Save the plot as a PNG file
+    plt.savefig('rhe_test.png', dpi=350)
+
+    # Close the plot to free up memory
+    plt.close(fig)
```

Comment:
Our figure tests work by returning the figure to display. It will then hash the final PNG and compare it that way. Our test suite will display the original PNG (if it has been uploaded) and it will show a comparison to the current one via a HTML page.

Reply:
Ok, so I'll just return the figure. I'm looking at the page at https://docs.sunpy.org/en/latest/dev_guide/contents/tests.html and it seems I'll need to do some other steps involving hashing the "correct" png, too.

Reply:
You can use the CI for that, circleci will output the hash you can copy and add that to the figure hash file. No need to run the test and hash the PNG yourself.

Reply:
Is there documentation for how to add things to the figure hash file? Thanks for your help!

Reply:
If you add the test, then on CircleCI, it will output the file. 

I think you have to login to CircleCI to see it though. 
If you see https://app.circleci.com/pipelines/github/sunpy/sunkit-image/1200/workflows/c613f21a-27d1-4a9a-9821-34ae0cfdb909/jobs/2964/artifacts

There is a [.tmp/py312-figure/figure_test_images/figure_hashes_mpl_390_ft_261_sunpy_513_astropy_610.json](https://output.circle-artifacts.com/output/job/c4af4d4d-645f-485d-b508-1207a21c61d8/artifacts/0/.tmp/py312-figure/figure_test_images/figure_hashes_mpl_390_ft_261_sunpy_513_astropy_610.json) which you can copy and paste the contents into the one in your branch. 

That should be everything you need to do.

Reply:
Ok, I've implemented this such that it passes the tests now (mostly, though I don't remember where to upload the baseline pngs). I also wrote an example gallery. That said, I was wondering if you thought that maybe this should go in "enhance" rather than "radial gradient filters," since it does more than just remove the radial falloff. I'm not sure where else to put a discussion like that at this point.

Reply:
Don't worry about the baseline figure, that gets uploaded by CircleCI when we merge this pull request.

Reply:
About where it should go is a good question, if it does more than just radial, then I would say that enhance is a better place for it. 

The best place for the discussion can be on the PR or the issue, it doesn't really matter to me.